src/main/java/com/pnudev/librarysystem/config/SecurityConfig.java

```diff
@@ -0,0 +1,55 @@
+package com.pnudev.librarysystem.config;
+
+import com.pnudev.librarysystem.security.JwtExceptionHandlerFilter;
+import com.pnudev.librarysystem.security.JwtTokenFilter;
+import lombok.AllArgsConstructor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.config.annotation.authentication.configuration.AuthenticationConfiguration;
+import org.springframework.security.config.annotation.method.configuration.EnableMethodSecurity;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+
+@AllArgsConstructor
+@Configuration
+@EnableWebSecurity
+@EnableMethodSecurity(securedEnabled = true)
+public class SecurityConfig {
+
+    private final JwtTokenFilter jwtTokenFilter;
+    private final JwtExceptionHandlerFilter jwtExceptionHandlerFilter;
+
+    @Bean
+    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+        http
+                .csrf().disable()
+                .formLogin().disable()
+                .httpBasic().disable()
+                .sessionManagement()
+                .sessionCreationPolicy(SessionCreationPolicy.STATELESS)
+                .and()
+                .addFilterAfter(jwtTokenFilter, UsernamePasswordAuthenticationFilter.class)
```

Comment:
So basically each request gets verified through JWT filters?

Reply:
Only those that need authentication: 
` if (authHeader == null || !authHeader.startsWith("Bearer "))  filterChain.doFilter(request, response);`
Or what do you mean?